compiler/rustc_data_structures/src/lib.rs

```diff
@@ -161,3 +162,8 @@ macro_rules! external_bitflags_debug {
         }
     };
 }
+
+/// `rustc_driver::main` installs a handler that will set this to `true` if
+/// the compiler has been sent a request to shut down, such as by a Ctrl-C.
+/// This static is placed here so that it is available to all parts of the compiler.
+pub static CTRL_C_RECEIVED: AtomicBool = AtomicBool::new(false);
```

Comment:
`rustc_data_structures` seems like a strange place to put this. You're using this in rustc_const_eval and rustc_driver_impl. rustc_driver_impl depends on rustc_const_eval. Can't you put the static into rustc_const_eval?

Reply:
Yup.